### DISCUSSION TITLE
Export/Import CSV - Devices doesn't work to import

### DISCUSSION THREAD

**Discussion by mroupie34** [netbox-community/netbox#13799] [General]
Hello everyone,

I need your help about this issue.
I got an error when i import the CSV. 

![image](https://github.com/netbox-community/netbox/assets/107417012/40e4fd90-bde1-4594-9b2e-53a03934cf48)

**Inside the CSV when i export them :** 

![image](https://github.com/netbox-community/netbox/assets/107417012/62bcdfce-f49c-42cd-93b5-c573ca7e4a67)

**Inside the CSV when i import them :** 

![image](https://github.com/netbox-community/netbox/assets/107417012/dd585436-0a61-4a7b-9c00-ffa9c7cbf666)
 
Do you know if this bug wil be resolved ?

What can I do while waiting ? Do you have a solution about that ?

**Version Netbox : 3.6.1**

Thanks a lot for yourfeedback,
Best regards,

Maxime ROUPIE
Engineer Systems and Network
Koesio Corporate IT

**Comment by candlerb** [comment]
Netbox's CSV Export and Import are completely different formats, and are not compatible with each other. This is documented in the [FAQ](https://github.com/netbox-community/netbox/wiki/Frequently-Asked-Questions#why-cant-i-pass-exported-data-directly-into-the-import-form).  So unfortunately, you can't use the export data to guide what correct import data might look like.

When you import some data from a CSV (even if it's the wrong format) you shouldn't be able to trigger a Python exception, so there *is* a bug here.

However, it's not possible to reproduce your bug from screenshots of spreadsheets. Please provide plain text than can be copy-pasted into the import box.

If you can't provide data which can reproduce the bug, then at least provide the backtrace (set DEBUG=True, and/or set ADMINS and EMAIL, in configuration.py)

There is a known issue where if you give unexpected column names, they are ignored. Hence if you give a column called (say) "Role" where it should be "role", it's as if you didn't give the column at all. See #11617 - although in future they won't be case sensitive, see #13775. This might be why your upload is failing and seeing "None" where it should see some real data.

For another example of the sorts of problems you can get with bad CSV data, see https://github.com/netbox-community/netbox/discussions/12259#discussioncomment-6923822

**Reply by mroupie34** [reply]
Hello,

Anyway, it's for exported "Rack" and then imported with all data.

For example my template : 

[Template-import-dbs-devices.csv](https://github.com/netbox-community/netbox/files/12669955/Template-import-dbs-devices.csv)

But i got an error : 

![image](https://github.com/netbox-community/netbox/assets/107417012/bc3dce80-74fd-43d1-b7f6-226363c711ef)

I don't know if it's possible or not because the nomenclature in Netbox is like this on the column "Rack" : 

![image](https://github.com/netbox-community/netbox/assets/107417012/6e5e24e8-9ba2-4016-8b7a-7add084ec0ea)


Thanks a lot for your feedback,
Max

**Reply by candlerb** [reply]
The allowed fields in a CSV upload are documented on the import page:
![image](https://github.com/netbox-community/netbox/assets/44789/3a4d4c6f-8c15-422a-913c-3da206b52730)

You've put column "rack_id" when it should be "rack". That's invalid, and as I explained before, is currently treated the same as if you'd left the column out completely. That's why Netbox is complaining that you didn't specify the rack.
